auth/service.cc

```diff
@@ -627,4 +632,63 @@ future<std::vector<permission_details>> list_filtered_permissions(
     });
 }
 
+future<> migrate_to_auth_v2(cql3::query_processor& qp, ::service::raft_group0_client& g0, abort_source& as) {
+    auto guard = co_await g0.start_operation(&as);
+    auto gen = [&qp, ts = guard.write_timestamp()] () -> mutations_generator {
+        for (const auto& cf_name : std::vector<sstring>{
+            "roles", "role_members", "role_attributes", "role_permissions"}) {
+            schema_ptr schema;
+            try {
+                schema = qp.db().find_schema(meta::legacy::AUTH_KS, cf_name);
+             } catch (data_dictionary::no_such_column_family&) {
+                continue; // some tables might not have been created if they were not used
+            }
+            auto rows = co_await qp.execute_internal(
+                format("SELECT * FROM {}.{}", meta::legacy::AUTH_KS, cf_name),
```

Comment:
Please name the columns explicitly, SELECT * is a misfeature since you can get more columns than you expect. The table is under our control so it's not a practical problem, but let's use clean coding practices.

Reply:
Oh I see below the code is dynamic and adjusts to the columns. But isn't the schema fixed?

Reply:
there are 4 different tables here which this code handles generically.

Reply:
> SELECT * is a misfeature

It's quite literal, here we want to migrate everything - so all columns should be used.